### ISSUE TITLE
SyntaxError? / Python 2.7 incompatibility?

### ISSUE THREAD

**Issue by TRIROG** [ngraziano/isystem-to-mqtt#2]
I'm getting an error:
pi@rPi ~ $ dump_isystem.py --serial /dev/ttyUSB0 --deviceid 10
  File "/usr/local/bin/dump_isystem.py", line 62
    print("{0:4d} => {1:5d} ".format(address, raw_values[index]), end='')
                                                                     ^
SyntaxError: invalid syntax

What could be the reason?

**Comment by ngraziano** [comment]
I test with Python 3.4, maybe it is the problem.

My complete setup script on my raspberry  :
````bash
virtualenv --system-site-packages -p /usr/bin/python3 env
source ./env/bin/activate
pip install --upgrade git+https://github.com/ngraziano/isystem-to-mqtt.git
````

**Comment by ngraziano** [comment]
I try to make it work with python 2.7. Tell me if it work.

**Comment by TRIROG** [comment]
Works like a charm!
BTW:
Is there any way to read offset set by remote control button  (-3*C to +3*C)?

**Comment by TRIROG** [comment]
Most likely python ver issue then - i'

**Comment by TRIROG** [comment]
dumping works fine now, polling however doesn't :\

pi@rPi ~/iSystem/isystem-to-mqtt/bin $ python poll_isystem_mqtt.py  <mqtt server ip>
Traceback (most recent call last):
  File "poll_isystem_mqtt.py", line 71, in <module>
    client.will_set(base_topic + "reading", "OFF", 1, True)
  File "/usr/local/lib/python2.7/dist-packages/paho/mqtt/client.py", line 1317, in will_set
    raise TypeError('payload must be a string, bytearray, int, float or None.')
TypeError: payload must be a string, bytearray, int, float or None.

Note , that even if i comment out the line 71 (i.e. `client.will_set(base_topic + "reading", "OFF", 1, True)`) I still get the following error:
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 552, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 505, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/local/lib/python2.7/dist-packages/paho/mqtt/client.py", line 2580, in _thread_main
    self.loop_forever(retry_first_connection=True)
  File "/usr/local/lib/python2.7/dist-packages/paho/mqtt/client.py", line 1378, in loop_forever
    rc = self.loop(timeout, max_packets)
  File "/usr/local/lib/python2.7/dist-packages/paho/mqtt/client.py", line 897, in loop
    rc = self.loop_read(max_packets)
  File "/usr/local/lib/python2.7/dist-packages/paho/mqtt/client.py", line 1177, in loop_read
    rc = self._packet_read()
  File "/usr/local/lib/python2.7/dist-packages/paho/mqtt/client.py", line 1766, in _packet_read
    rc = self._packet_handle()
  File "/usr/local/lib/python2.7/dist-packages/paho/mqtt/client.py", line 2245, in _packet_handle
    return self._handle_connack()
  File "/usr/local/lib/python2.7/dist-packages/paho/mqtt/client.py", line 2306, in _handle_connack
    self.on_connect(self, self._userdata, flags_dict, result)
  File "poll_isystem_mqtt.py", line 84, in on_connect
    the_client.subscribe(subscribe_list)
  File "/usr/local/lib/python2.7/dist-packages/paho/mqtt/client.py", line 1112, in subscribe
    raise ValueError('Invalid topic.')
ValueError: Invalid topic.

Traceback (most recent call last):
  File "poll_isystem_mqtt.py", line 138, in <module>
    read_zone(zone[0],zone[1])
  File "poll_isystem_mqtt.py", line 113, in read_zone
    tag_definition.publish(client, base_topic, raw_values, index)
  File "/usr/local/lib/python2.7/dist-packages/isystem_to_mqtt/tag_definition.py", line 26, in publish
    client.publish(base_topic + self.tag_name, value, retain=True)
  File "/usr/local/lib/python2.7/dist-packages/paho/mqtt/client.py", line 968, in publish
    raise TypeError('payload must be a string, bytearray, int, float or None.')
TypeError: payload must be a string, bytearray, int, float or None.

Maybe another Python 2.7 incompatibility?

**Comment by ngraziano** [comment]
I remove an incorrect compatibility import for python 2.7. It should work now.

For the offset set by remote control button, i do not find a way to read it, when you set offset the value read stay the same.
I also can't set the end day for vacation (I can read it but if I set it it does not work).

If you find interesting MODBUS address in the system, let me know.

**Comment by TRIROG** [comment]
Works, well partially at least:
I can see some messages get through to my MQTT server but then i get an error:

Traceback (most recent call last):
  File "poll_isystem_mqtt.py", line 137, in <module>
    read_zone(zone[0],zone[1])
  File "poll_isystem_mqtt.py", line 112, in read_zone
    tag_definition.publish(client, base_topic, raw_values, index)
  File "/usr/local/lib/python2.7/dist-packages/isystem_to_mqtt/tag_definition.py", line 25, in publish
    client.publish(base_topic + self.tag_name, value, retain=True)
  File "/usr/local/lib/python2.7/dist-packages/paho/mqtt/client.py", line 968, in publish
    raise TypeError('payload must be a string, bytearray, int, float or None.')
TypeError: payload must be a string, bytearray, int, float or None.

After enabling the --DEBUG option this is what i get:

DEBUG:isystem_to_mqtt.tag_definition:value zone-b/mode  = Automatique
DEBUG:isystem_to_mqtt.tag_definition:value zone-b/mode-raw  = 8
DEBUG:isystem_to_mqtt.tag_definition:value zone-b/mode-simple  = Automatique
DEBUG:isystem_to_mqtt.tag_definition:value zone-b/sensor-influence  = 3
DEBUG:isystem_to_mqtt.tag_definition:value zone-a/antifreeze-duration  = 0
DEBUG:isystem_to_mqtt.tag_definition:value zone-b/antifreeze-duration  = 0
DEBUG:isystem_to_mqtt.tag_definition:value boiler/heating-power  = 65600600535
Traceback (most recent call last):
  File "poll_isystem_mqtt.py", line 137, in <module>
    read_zone(zone[0],zone[1])
  File "poll_isystem_mqtt.py", line 112, in read_zone
    tag_definition.publish(client, base_topic, raw_values, index)
  File "/usr/local/lib/python2.7/dist-packages/isystem_to_mqtt/tag_definition.py", line 25, in publish
    client.publish(base_topic + self.tag_name, value, retain=True)
  File "/usr/local/lib/python2.7/dist-packages/paho/mqtt/client.py", line 968, in publish
    raise TypeError('payload must be a string, bytearray, int, float or None.')
TypeError: payload must be a string, bytearray, int, float or None.

Seems like there is a problem with the:
DEBUG:isystem_to_mqtt.tag_definition:value boiler/heating-power  = 65600600535
or a definition starting at 788 or later?

Changing the  ZONE_TABLE_MODULENS_O definition in tables.py to
```
ZONE_TABLE_MODULENS_O = [(231, 20),
                         (507, 4),
                         (471, 10),
                         (600, 21),
                         (637, 24),
                         (721, 4),]
                         #(788, 9),]
```
Solves the issue.


It would be great if the script wouldn't demand TLS by default (my MQTT server is internal only and i therefore use no authentication or encryption)

Where is the polling interval defined?

**Comment by ngraziano** [comment]
Can you send me the result of the dump_isystem.py command for the address from 780 to 800 ?

In my boiler (Modulens-O AFC 30)  address  788 ,789 ,790 are MWh, KWh and Wh part of the total heat consumed since first start.

For the TLS I don't understand the problem. If you don't put the "--tls12" flag it doesn't use TLS (the same for --user and --password).

The polling interval is set by "--interval xx" value. But if you write some value (like heating/zone-a/day-target-temperature/SET) it does not wait for the interval and read immediately.

**Comment by TRIROG** [comment]
I'm using Innovens MCA25 with iSystem 
Here is the dump:
```
 780 =>     0
 781 =>     0
 782 =>     0
 783 => 65535
 784 =>     0
 785 =>     0
 786 => 65535
 787 =>     0
 788 => 65535   boiler/heating-power    65600600535
 789 => 65535 ^
 790 => 65535 ^
 791 => 65535   boiler/dhw-power        65600600535
 792 => 65535 ^
 793 => 65535 ^
 794 => 65535   boiler/cooling-power    65600600535
 795 => 65535 ^
 796 => 65535 ^
 797 => 65535
 798 => 65535
 799 => 65535
 800 =>     0
```


Regarding the TLS:
```
pi@rPi ~/iSystem/isystem-to-mqtt/bin $ python poll_isystem_mqtt.py --log DEBUG 192.168.0.249
Traceback (most recent call last):
  File "poll_isystem_mqtt.py", line 30, in <module>
    action="store_const", const=ssl.PROTOCOL_TLSv1_2)
AttributeError: 'module' object has no attribute 'PROTOCOL_TLSv1_2'
```
but if i comment out the 
```
#parser.add_argument("--tls12", help="use TLS 1.2", dest="tls",
#                    action="store_const", const=ssl.PROTOCOL_TLSv1_2)
```
and
```

#if args.tls:
#    client.tls_set(args.cacert, tls_version=args.tls)
#    port_mqtt = 8883
```

Script runs just fine.

Regarding the interval: 
d0h (feeling silly)

**Comment by TRIROG** [comment]
Hmmm ... address 787 should  be "1" to enable data about heat consumation but in my case it seems to be set to "0" ...
Will try and change this bit ...

**Comment by ngraziano** [comment]
I try to handle the 65535 values with a None value when counter are not available.

For the ssl I try to handle the missing constant but it is a workaround, I will change this code in future.

**Comment by TRIROG** [comment]
I still get an error :
Traceback (most recent call last):
  File "poll_isystem_mqtt.py", line 73, in <module>
    if args.tls:
AttributeError: 'Namespace' object has no attribute 'tls'

but adding `try:` before line 73 fixes the issue.
```
try:
    if args.tls:
        _LOGGER.debug("Set TLS mode.")
        client.tls_set(args.cacert, tls_version=args.tls)
        port_mqtt = 8883
except:
    pass
```
The address 788-796 now also normally resolves to "None"

Thanks!

**Comment by TRIROG** [comment]
Fixed in 19eea8f.
